bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt

```diff
@@ -13,6 +16,15 @@ class LazyIrohaAPIPool : Closeable {
 
     // Collection of IrohaAPIs
     private val apiPool = HashMap<IrohaConfig, IrohaAPI>()
+    // Executor service that manages old and stale IrohaAPIs
+    private val scavengerExecutorService = Executors.newSingleThreadScheduledExecutor(
+        ThreadFactoryBuilder().setNameFormat("iroha-api-scavenger").build()
+    )
+
+    init {
+        // Removes old APIs every 5 minutes
```

Comment:
What if the connection dies in < 5 minutes? Or it does not matter from `IrohaAPI` perspective?

Reply:
Nothing bad will happen, probably. If you accidentally took stale IrohaAPI, new API will be created.